CMSIS/NN/Source/PoolingFunctions/arm_avgpool_s8.c

```diff
@@ -228,51 +248,101 @@ arm_status arm_avgpool_s8(const int dim_src_height,
+          q7_t *start = src + ch_src * (k_x + base_idx_x + (k_y + base_idx_y) * dim_src_width);
+
+          if (count == 0)
+          {
+            arm_q7_to_q15_no_shift(start, bufferA, ch_src);
+          }
+          else
+          {
+            arm_nn_accumulate_q7_to_q15(bufferA, start, ch_src);
+          }
+          count++;
+        }
+      }
+      buffer_scale_back_q15_to_q7_and_clamp(bufferA, dst, ch_src, count, act_min, act_max);
+      dst += ch_src;
+    }
+  }
+#else
+
+  /* Reference C code adapted from CMSIS-NN arm_avepool_q7_HWC.
+   */
+  (void)bufferA;
+  int16_t i_ch_in, i_x, i_y;
+  int16_t k_x, k_y;
+
+  for (i_y = 0; i_y < dim_dst_height; i_y++)
+  {
+    for (i_x = 0; i_x < dim_dst_width; i_x++)
+    {
+      for (i_ch_in = 0; i_ch_in < ch_src; i_ch_in++)
+      {
+        int sum = 0;
+        int count = 0;
+        for (k_y = i_y * stride_height - padding_height; k_y < i_y * stride_height - padding_height + dim_kernel_height; k_y++)
+        {
+          for (k_x = i_x * stride_width - padding_width; k_x < i_x * stride_width - padding_width + dim_kernel_width; k_x++)
+          {
+            if (k_y >= 0 && k_x >= 0 && k_y < dim_src_height && k_x < dim_src_width)
             {
-                int       sum = 0;
-                int       count = 0;
-                for (k_y = i_y * stride_height - padding_height; k_y < i_y * stride_height - padding_height + dim_kernel_height; k_y++)
-                {
-                    for (k_x = i_x * stride_width - padding_width; k_x < i_x * stride_width - padding_width + dim_kernel_width; k_x++)
-                    {
-                        if (k_y >= 0 && k_x >= 0 && k_y < dim_src_height && k_x < dim_src_width)
-                        {
-                            sum += src[i_ch_in + ch_src * (k_x + k_y * dim_src_width)];
-                            count++;
-                        }
-                    }
-                }
-                sum = sum > 0 ? (sum + count / 2) / count : (sum - count / 2) / count;
-                sum = MAX(sum, act_min);
-                sum = MIN(sum, act_max);
-
-                dst[i_ch_in + ch_src * (i_x + i_y * dim_dst_width)] = sum;
+              sum += src[i_ch_in + ch_src * (k_x + k_y * dim_src_width)];
+              count++;
             }
+          }
         }
+        sum = sum > 0 ? (sum + count / 2) / count : (sum - count / 2) / count;
+        sum = MAX(sum, act_min);
+        sum = MIN(sum, act_max);
+
+        dst[i_ch_in + ch_src * (i_x + i_y * dim_dst_width)] = sum;
+      }
     }
+  }
 
-    return ARM_MATH_SUCCESS;
+#endif
+  return ARM_MATH_SUCCESS;
 }
 
 #endif /* ARM_MATH_MVEI */
 
 int32_t arm_avgpool_s8_get_buffer_size(const int dim_dst_width,
                                        const int ch_src)
 {
-    (void)dim_dst_width;
-    (void)ch_src;
-    return 0;
+#if defined(ARM_MATH_DSP) && !defined(ARM_MATH_MVEI)
+  return (ch_src * dim_dst_width);
```

Comment:
Please fix to the right amount of memory needed.

Reply:
Yes. will do